easy-ui-react/src/Button/Button.module.scss

```diff
@@ -0,0 +1,321 @@
+  }
+
+  &:disabled {
+    color: var(--ezui-color-gray-300);
+  }
+
+  &:active:not(:disabled) {
+    color: var(--ezui-color-purple-700);
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    color: var(--ezui-color-purple-600);
+  }
+}
+
+.colorSuccess.variantFilled {
+  color: var(--ezui-color-blue-800);
+
+  &:focus {
+    background-color: var(--ezui-color-green-600);
+  }
+
+  &:not(:disabled):active {
+    color: var(--ezui-color-white);
+    background-color: var(--ezui-color-green-700);
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    background-color: var(--ezui-color-green-600);
+  }
+}
+
+.colorSuccess.variantOutlined {
+  &:active:not(:disabled) {
+    color: var(--ezui-color-green-700);
+    border-color: var(--ezui-color-green-700);
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    color: var(--ezui-color-green-600);
+    border-color: var(--ezui-color-green-600);
+  }
+}
+
+.colorWarning.variantFilled {
+  color: var(--ezui-color-blue-800);
+  &:focus {
+    background-color: var(--ezui-color-red-600);
+  }
+
+  &:not(:disabled):active {
+    background-color: var(--ezui-color-red-700);
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    background-color: var(--ezui-color-red-600);
+  }
+}
+
+.colorWarning.variantOutlined {
+  &:active:not(:disabled) {
+    color: var(--ezui-color-red-700);
+    border-color: var(--ezui-color-red-700);
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    color: var(--ezui-color-red-600);
+    border-color: var(--ezui-color-red-600);
+  }
+}
+
+.colorNeutral.variantFilled {
+  &:focus {
+    background-color: var(--ezui-color-black);
+  }
+
+  &:not(:disabled):active {
+    background-color: var(--ezui-color-blue-700);
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    background-color: var(--ezui-color-black);
+  }
+}
+
+.colorSupport.variantOutlined {
+  &:active:not(:disabled) {
+    color: var(--ezui-color-gray-700);
+    border-color: var(--ezui-color-gray-700);
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    color: var(--ezui-color-black);
+    border-color: var(--ezui-color-black);
+  }
+}
+
+.sizeMd.variantFilled,
+.sizeMd.variantOutlined,
+.sizeMd.variantLink {
+  @include font-style("button");
+}
+
+.sizeSm.variantFilled,
+.sizeSm.variantOutlined,
+.sizeSm.variantLink {
+  @include font-style("small-button");
+}
+
+.sizeMd.variantFilled {
+  padding: var(--ezui-space-1-5) var(--ezui-space-2);
+}
+
+.sizeMd.variantOutlined,
+.sizeMd.variantLink {
+  padding: calc(#{var(--ezui-space-1-5)} - #{var(--ezui-shape-border-width-1)})
+    calc(#{var(--ezui-space-2)} - #{var(--ezui-shape-border-width-1)});
+}
+
+.sizeSm.variantFilled {
+  padding: var(--ezui-space-1);
+}
+
+.sizeSm.variantOutlined,
+.sizeSm.variantLink {
+  padding: calc(#{var(--ezui-space-1)} - #{var(--ezui-shape-border-width-1)});
+}
+
+.ButtonBlock {
+  display: flex;
+  width: 100%;
+}
+
+.StartIcon {
+  margin-left: var(--ezui-space-1);
+}
+
+.EndIcon {
+  margin-right: var(--ezui-space-1);
+}
```

Comment:
with CSS modules, i tend to see classes written in more typical CSS syntax and without the root modifier. in this case, i think these could become:

```
.block {}

.iconAtStart {}

.iconAtEnd {}
```